server/src/api/event_filter.go

```diff
@@ -0,0 +1,238 @@
+	"github.com/gorilla/websocket"
+	dto2 "scrumlr.io/server/common/dto"
+	"scrumlr.io/server/database/types"
+	"scrumlr.io/server/logger"
+	"scrumlr.io/server/realtime"
+)
+
+func isModerator(clientID uuid.UUID, sessions []*dto2.BoardSession) bool {
+	for _, session := range sessions {
+		if clientID == session.User.ID {
+			if session.Role == types.SessionRoleModerator || session.Role == types.SessionRoleOwner {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// ToDo: Refactor the client sorting or even find a better solution to avoid it
+type client struct {
+	userID uuid.UUID
+	conn   *websocket.Conn
+	role   types.SessionRole
+}
+
+type ByRole []client
+
+func (b ByRole) Len() int {
+	return len(b)
+}
+
+func (b ByRole) Less(i, j int) bool {
+	return b[i].role < b[j].role
+}
+
+func (b ByRole) Swap(i, j int) {
+	b[i], b[j] = b[j], b[i]
+}
+
+func sortClientsByRole(clients map[uuid.UUID]*websocket.Conn, boardParticipants []*dto2.BoardSession) []client {
+	mappedClients := make([]client, 0, len(clients))
+	for id := range clients {
+		var role types.SessionRole
+		// retrieve role
+		for _, user := range boardParticipants {
+			if id == user.User.ID {
+				role = user.Role
+			}
+		}
+		aClient := client{
+			userID: id,
+			conn:   clients[id],
+			role:   role,
+		}
+		mappedClients = append(mappedClients, aClient)
+	}
+
+	// Sort new slice via Role and iterate over it
+	sort.Sort(ByRole(mappedClients))
+	return mappedClients
+}
+
+func parseColumnUpdatedEvent(data interface{}) ([]*dto2.Column, error) {
+	var ret []*dto2.Column
+	b, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &ret)
+	if err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+func parseNotesUpdated(data interface{}) ([]*dto2.Note, error) {
+	var ret []*dto2.Note
+
+	b, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &ret)
+	if err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+func parseBoardUpdated(data interface{}) (*dto2.Board, error) {
+	var ret *dto2.Board
+
+	b, err := json.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &ret)
+	if err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+func (boardSubscription *BoardSubscription) eventFilter(event *realtime.BoardEvent, userID uuid.UUID) *realtime.BoardEvent {
+	isMod := isModerator(userID, boardSubscription.boardParticipants)
+	if event.Type == realtime.BoardEventColumnsUpdated {
+		columns, err := parseColumnUpdatedEvent(event.Data)
+		if err != nil {
+			logger.Get().Errorw("unable to parse columnUpdated in event filter", "board", boardSubscription.boardSettings.ID, "session", userID, "error", err)
+		}
+		// Cache the incoming changes, mod only since they receive all changes
+		if isMod {
+			boardSubscription.boardColumns = columns
+			return event
+		}
+
+		var ret realtime.BoardEvent
+		var seeableColumns = make([]*dto2.Column, 0, len(boardSubscription.boardColumns))
```

Comment:
is _seeable_ an actual word? 👀 I think visible is more appropiate

Reply:
Changed it to "visible" 😁